flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackend.java

```diff
@@ -245,7 +249,8 @@
 		TaskKvStateRegistry kvStateRegistry,
 		TtlTimeProvider ttlTimeProvider,
 		MetricGroup metricGroup,
-		@Nonnull Collection<KeyedStateHandle> stateHandles) throws Exception;
+		@Nonnull Collection<KeyedStateHandle> stateHandles,
```

Comment:
This annotation should ideally be propagated to all implementations, so that they are also annotated as `@Nonnull`. Your IDE should be able to do this for you as an automated quick fix function.